index.d.ts

```diff
@@ -22,3 +22,27 @@ console.log(await fn(2));
 ```
 */
 export default function makeAsynchronous<T extends AnyFunction>(function_: T): Asyncify<T>;
+
+type IteratorFunctionValue<T> = T extends (...arguments_: any) => AsyncIterable<infer Value> | Iterable<infer Value> ? Value : any;
+
+/**
+Make the iterator returned by a function asynchronous by running it in a worker.
+
+Returns a wrapped version of the given function which executes asynchronously in a background thread (meaning it will not block the main thread).
+
+The given function is serialized, so you cannot use any variables or imports from outside the function scope. You can instead pass in arguments to the function.
+
+@example
+```
+import {makeAsynchronousIterator} from 'make-asynchronous';
+
+const fn = makeAsynchronousIterator(function * () {
+	yield * performExpensiveOperation(number);
+});
+
+for await (const number of await fn(2)) {
+	console.log(number);
+}
+```
+*/
+export function makeAsynchronousIterator<T extends (...arguments_: any) => AsyncIterable<any> | Iterable<any>>(function_: T): SetReturnType<T, Promise<AsyncIterable<IteratorFunctionValue<T>>>>;
```

Comment:
I don't think `Promise` is correct here.

```suggestion
export function makeAsynchronousIterator<T extends (...arguments_: any) => AsyncIterable<any> | Iterable<any>>(function_: T): SetReturnType<T,AsyncIterable<IteratorFunctionValue<T>>>;
```